controllers/solrprometheusexporter_controller.go

```diff
@@ -434,53 +458,133 @@ func (r *SolrPrometheusExporterReconciler) buildSecretWatch(secretField string,
 		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})), nil
 }
 
-func (r *SolrPrometheusExporterReconciler) reconcileTLSConfig(prometheusExporter *solrv1beta1.SolrPrometheusExporter) (*util.TLSConfig, error) {
-	opts := prometheusExporter.Spec.SolrReference.SolrTLS
-
-	tls := &util.TLSConfig{}
-	tls.InitContainerImage = prometheusExporter.Spec.BusyBoxImage
-	tls.Options = opts
+// Reconcile the various options for configuring TLS for the exporter
+// The exporter is a client to Solr pods, so can either just have a truststore so it trusts Solr certs
+// Or it can have its own client auth cert when Solr mTLS is required
+func (r *SolrPrometheusExporterReconciler) reconcileTLSConfig(prometheusExporter *solrv1beta1.SolrPrometheusExporter) (*util.TLSCerts, error) {
+	tls := util.TLSCertsForExporter(prometheusExporter)
+	opts := tls.ClientConfig.Options
 
 	if opts.PKCS12Secret != nil {
```

Comment:
Looks like there's a lot of duplicated code between `SolrPrometheusExport.solrTLS`, `SolrCloud.solrTLS` and `SolrCloud.solrClientTLS`. Could it be one generic method used for all of them, or is that not feasible?

Reply:
yeah, let me take another stab on consolidating the reconcile loop around this, definitely a lot of similarity that can be reused.

Reply:
Greatly reduced the amount of code in the TLS reconcile loops for SolrCloud and Exporter in the last commit. Good eye!